sympy/solvers/polysys.py

```diff
@@ -429,3 +432,567 @@ def solve_triangulated(polys, *gens, **args):
 
         solutions = _solutions
     return sorted((s for s, _ in solutions), key=default_sort_key)
+
+
+def solve_poly_system_cad(seq, gens, return_one_sample=True):
+    """
+    Solves a system of polynomial inequalities/equalities via
+    cylindrical algebraic decomposition. Returns a sample point
+    in one (if return_one_sample=True) or all cells over which
+    the system holds. If the system is invalid over all cells, then
+    we return None.
+
+    Parameters
+    ==========
+
+    seq: a list/tuple/set
+        Listing all the (in)equalities that are needed to be solved
+    gens: generators
+        Generators of the (in)equalities in seq for which we want the
+        solutions
+    return_one_sample: bool
+        If True, returns a single satisfying point. If False, returns
+        a sample point from each CAD cell over which the system holds.
+
+
+    Returns
+    =======
+
+    List[Dict]
+        a list of dicts with the returned sample points. Each dict
+        is a point, with the keys being the variables. If the system
+        is unsatisfiable, then an empty list is returned.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x,y
+    >>> from sympy.solvers.polysys import solve_poly_system_cad
+
+    >>> solve_poly_system_cad([-x**2-1 > 0], [x])
+    []
+
+    >>> solve_poly_system_cad([x**2-1 > 0], [x])
+    [{x: -2}]
+
+    >>> solve_poly_system_cad([x**2-1 > 0], [x], False)
+    [{x: -2}, {x: 2}]
+
+    >>> solve_poly_system_cad([y*x**2>0, x+y<1], [x,y], False)
+    [{x: -1, y: 1/2}, {x: 1/4, y: 1/2}, {x: -1, y: 1}, {x: -2, y: 2}]
+    """
+    # prepare the atoms
+    atoms = [Poly(p.lhs - p.rhs, gens) for p in seq]
+
+    sample_points = cylindrical_algebraic_decomposition(atoms, gens)
+    valid_samples = []
+
+    for sample in sample_points:
+        if all(expr.subs(sample) for expr in seq):
+            valid_samples.append(sample)
```

Comment:
There are a couple of problems with this approach.

Firstly the sample points can contain floats and so this depends on testing exact equality of floats which does not generally work. Really floats should not be used.

Secondly this is trying to infer after the fact what the sign was of each polynomial when this information is already known during each lifting step when computing the cylindrical algebraic decomposition. It would be better to keep track of the signs of the polynomials during lifting.